tests/test_cli_compile.py

```diff
@@ -3002,13 +3002,13 @@ def test_raise_error_on_unknown_config_option(
     out = runner.invoke(cli, [req_in.as_posix(), "--config", config_file.as_posix()])
 
     assert out.exit_code == 2
-    assert "No such config key 'unknown_option'" in out.stderr
+    assert "No such config key 'unknown-option'" in out.stderr
 
 
 def test_raise_error_on_invalid_config_option(
     pip_conf, runner, tmp_path, make_config_file
 ):
-    config_file = make_config_file("dry-run", ["invalid", "value"])
+    config_file = make_config_file("dry_run", ["invalid", "value"])
```

Comment:
```
dry_run = false 
dry-run = false
```

Shouldn't we support both variants?

Reply:
Do you have suggestion how to do this with `pytest`? We could also make the `make_config_file()` function accept multiple parameters, BTW.

Reply:
This might take some refactoring. Let's address that in a following-up PR.